### ISSUE TITLE
Describe the build environment for conda-recipes

### ISSUE THREAD

**Issue by yesimon** [conda/conda-recipes#505]
Based off the discussion here ContinuumIO/anaconda-issues#577

I think it would be useful to describe the build environment for default channel conda packages in the online documentation or README (things like OS, gcc version / clang version etc.) This would be useful for recipe authors to avoid any binary incompatibilities due to changing compiler version etc.

**Comment by ccordoba12** [comment]
@groutr, @msarahan, what do you think about this one?

**Comment by msarahan** [comment]
I think that's a great idea.  I don't know the information for Linux or Mac, though.

For Windows:
Build machine: Win 10 x64
Python 2.7: VS 2008 Pro, SP1
Python 3.3, 3.4: VS 2010
Python 3.5+: VS 2015

Win 7 SDK and Win 10 SDK (7 SDK in particular has some older build scripts that have been removed.)

Not terribly important, more of an execution detail:
Git for Windows

**Comment by bjodah** [comment]
@msarahan pointed me to: https://github.com/ContinuumIO/docker-images/tree/master/conda_builder_linux

I think it would serve as a good reference environment for linux-64?

**Comment by msarahan** [comment]
Yes, great idea.  That was my hope, anyway: to provide a "standard" docker image that anyone could readily use to be on the same page.

This image can also build linux-32 packages.  You just need to use a different start script from that repo (or set the variables that the start script passes in).

**Comment by yesimon** [comment]
Linking in ContinuumIO/anaconda-issues#413 to describe the OS X build environment, which I think is at minimum OS X version 10.7 yet linking with libstdc++? This is based on 8c1d6f475d103562b4d635bac20cffc88beb809e as well.

Would it make more sense to move to libc++ because the earliest it's available is 10.7, also making it the transition to 10.9 easier where it becomes the default?
